localstack/services/install.py

```diff
@@ -303,31 +304,46 @@ def install_elasticmq():
 
 def install_kinesis():
     if config.KINESIS_PROVIDER == "kinesalite":
-        return install_kinesalite()
-    elif config.KINESIS_PROVIDER == "kinesis-mock":
-        return install_kinesis_mock()
-    else:
-        raise ValueError("unknown kinesis provider %s" % config.KINESIS_PROVIDER)
+        install_kinesalite()
+        return
+    if config.KINESIS_PROVIDER == "kinesis-mock":
+        is_installed, bin_path = get_is_kinesis_mock_installed()
+        if not is_installed:
+            install_kinesis_mock(bin_path)
+        return
+    raise ValueError("unknown kinesis provider %s" % config.KINESIS_PROVIDER)
+
+
+def _apply_patches_kinesalite():
+    files = [
+        "%s/kinesalite/validations/decreaseStreamRetentionPeriod.js",
+        "%s/kinesalite/validations/increaseStreamRetentionPeriod.js",
+    ]
+    for file_path in files:
+        file_path = file_path % INSTALL_DIR_NPM
+        replace_in_file("lessThanOrEqual: 168", "lessThanOrEqual: 8760", file_path)
 
 
 def install_kinesalite():
     if not os.path.exists(INSTALL_PATH_KINESALITE_CLI):
         log_install_msg("Kinesis")
         run('cd "%s" && npm install' % MODULE_MAIN_PATH)
+        _apply_patches_kinesalite()
 
 
-def install_kinesis_mock():
-    target_dir = INSTALL_PATH_KINESIS_MOCK
-
+def get_is_kinesis_mock_installed() -> Tuple[bool, str]:
+    """
+    Checks the host system to see if kinesis mock is installed and where.
+    Returns:
+        True if kinesis mock is installed (False otherwise) and the expected installation path
+    """
```

Comment:
I'm not sure what the overall feeling at LocalStack is around docstrings. I don't see many of them in the codebase, but I find them helpful (and was missing them during while working on this refactor) so I took the time to include some here.
If these are unwanted or should be formatted differently LMK and I'll fix it :)

Reply:
Thanks for teaching me that the name of that format is reST! We used that at my previous employer and I like it better than what I put up here, which is just based on the first example that I pulled up :)